modules/gateway/nodes.go

```diff
@@ -43,15 +44,26 @@ func (g *Gateway) removeNode(addr modules.NetAddress) error {
 	return nil
 }
 
+// randomNode returns a random node from the gateway. An error can be returned
 func (g *Gateway) randomNode() (modules.NetAddress, error) {
-	if len(g.nodes) > 0 {
-		r, _ := crypto.RandIntn(len(g.nodes))
-		for node := range g.nodes {
-			if r <= 0 {
-				return node, nil
-			}
-			r--
+	if len(g.nodes) == 0 {
+		return "", errNoPeers
+	}
+
+	// Select a random peer. Note that the algorithm below is roughly linear in
+	// the number of nodes known by the gateway, and this number can approach
+	// every node on the network. If the network gets large, this algorithm
+	// will either need to be refactored, or more likely a cap on the size of
+	// g.nodes will need to be added.
+	r, err := crypto.RandIntn(len(g.nodes))
+	if err != nil {
+		return err
+	}
+	for node := range g.nodes {
+		if r <= 0 {
+			return node, nil
 		}
+		r--
 	}
 
 	return "", errNoPeers
```

Comment:
this extra return statement is unnecessary, since `r` is less than `len(g.nodes)` by definition, so the function will return at some point in the for loop